src/scheduled_tasks.rs

```diff
@@ -323,62 +328,65 @@ fn update_banned_when_expired(conn: &mut PgConnection) {
 }
 
 /// Updates the instance software and version
-fn update_instance_software(conn: &mut PgConnection, user_agent: &str) {
+///
+/// TODO: this should be async
+/// TODO: if instance has been dead for a long time, it should be checked less frequently
```

Comment:
For this one, in the select below, you could do `let instances = instance::table.filter(coalesce(updated, published).gt(now - 1.months())`

Even better, would be to add this as `alive_instances` in impls/instance.rs

Another possibility, would be to recheck the alive_instances every day, but only re-check all of them (even previously dead ones) every month. Up to you.

Reply:
Or check old instances using random probability, eg in 1% of all checks.

Reply:
Anyway this can be improved later, no need to include it in this PR.